### DISCUSSION TITLE
ROS2 release team for GTSAM

### DISCUSSION THREAD

**Discussion by jlblancoc** [borglab/gtsam#1420] [General]
Current ROS build farm status for GTSAM: 

| ROS distro | develop CI | Stable release | Next build CI |
|---|---|---|---|
| ROS 1 Noetic |  [![Build Status](https://build.ros.org/job/Ndev__gtsam__ubuntu_focal_amd64/badge/icon)](https://build.ros.org/job/Ndev__gtsam__ubuntu_focal_amd64/) | [![Version](https://img.shields.io/ros/v/noetic/gtsam)](https://index.ros.org/search/?term=gtsam) | - |
| ROS 2 Humble | [![Build Status](https://build.ros2.org/job/Hdev__gtsam__ubuntu_jammy_amd64/badge/icon)](https://build.ros2.org/job/Hdev__gtsam__ubuntu_jammy_amd64/) | [![Version](https://img.shields.io/ros/v/humble/gtsam)](https://index.ros.org/search/?term=gtsam) | [![Build Status](https://build.ros2.org/job/Hbin_uJ64__gtsam__ubuntu_jammy_amd64__binary/badge/icon)](https://build.ros2.org/job/Hbin_uJ64__gtsam__ubuntu_jammy_amd64__binary/) |
| ROS 2 Iron | [![Build Status](https://build.ros2.org/job/Idev__gtsam__ubuntu_jammy_amd64/badge/icon)](https://build.ros2.org/job/Idev__gtsam__ubuntu_jammy_amd64/) | [![Version](https://img.shields.io/ros/v/iron/gtsam)](https://index.ros.org/search/?term=gtsam) | [![Build Status](https://build.ros2.org/job/Ibin_uJ64__gtsam__ubuntu_jammy_amd64__binary/badge/icon)](https://build.ros2.org/job/Ibin_uJ64__gtsam__ubuntu_jammy_amd64__binary/) |
| ROS 2 Rolling | [![Build Status](https://build.ros2.org/job/Rdev__gtsam__ubuntu_jammy_amd64/badge/icon)](https://build.ros2.org/job/Rdev__gtsam__ubuntu_jammy_amd64/) | [![Version](https://img.shields.io/ros/v/rolling/gtsam)](https://index.ros.org/search/?term=gtsam) | [![Build Status](https://build.ros2.org/job/Rbin_uJ64__gtsam__ubuntu_jammy_amd64__binary/badge/icon)](https://build.ros2.org/job/Rbin_uJ64__gtsam__ubuntu_jammy_amd64__binary/) |

-----
(Original post)

Hi all: 

I'm willing to help GTSAM to become accessible to ROS2 developers via precompiled binary releases, so it can be listed as any other dependencies in a `package.xml` as `<depend>gtsam</depend>` and/or installed via `sudo apt install ros-${ROS_DISTRO}-gtsam`.

In order to do so, we need to add a ROS2 release repository via https://github.com/ros2-gbp
Examples: [1](https://github.com/ros2-gbp/ros2-gbp-github-org/issues/159), or [2](https://github.com/ros2-gbp/ros2-gbp-github-org/issues/159). For those unfamiliar with the ROS2 release mechanism, basically it's based on GBP repos which *must* belong to the ros2-gbp GitHub Organization, and they have this protocol to request the creation of new release teams and gbp repos via issue tickets.

My proposal is to request the creation of:
- a release team: `borglab`
- repositories: `gtsam`
- Team members: ... who wants to be added? :-) 

Team members are those with the responsibility of running `bloom-release` with each stable release of gtsam (or non-stable releases if some critical error is fixed related to building within a ROS2 environment).
I would be glad to be in that team, but would be great if at least one or two other developers are added too.

**Comment by ProfFan** [comment]
I am open to doing it, but should we wait until boost is removed?

**Reply by jlblancoc** [reply]
Cool. There's no need, actually.... Unless develop is too unstable right now, is that the case?? 

 The plan would be like:
- [x] create the release repo to officially having everything set up.
- [x] add gtsam as 'source ' only first in the rolling distro. That will become a new CI builder. For each commit to the development branch (develop, I guess) a build will be attempted at their build servers. I could add another branch as 'stable' if develop is too unstable at present (??)
- [x] once builds are ok (one never knows what funny errors may show up), then, create the first official Ros release using bloom-release, pushing to all active ros2 distros.

**Reply by dellaert** [reply]
I am planning to remove boost completely this weekend, and I'd prefer to wait, and then get rid of build errors until it succeeds, and tag that 4.3a0, and we can pin ROS to 4.3a0?

**Comment by dellaert** [comment]
I am very happy this will happen, as roboticsbook.org could use a ROS example, as well.

**Comment by jlblancoc** [comment]
Rolling status:  [![Build Status](https://build.ros2.org/job/Rbin_uJ64__gtsam__ubuntu_jammy_amd64__binary/1/badge/icon)](https://build.ros2.org/job/Rbin_uJ64__gtsam__ubuntu_jammy_amd64__binary/1/)

It builds OK for ROS 2 Rolling, Yay! 
So, PRs have been opened now to add gtsam to  [Humble](https://github.com/ros/rosdistro/pull/36344) and [Foxy](https://github.com/ros/rosdistro/pull/36343) too.

Note that a "distro sync" must happen before users can do `sudo apt install ros-$ROS_DISTRO-gtsam`, but we are almost there... Also, gtsam would finally be able to be listed as a dependency in other packages instead of embedded as a submodule or alike.

I'm assuming ROS 1 releases are almost obsolete at this point and didn't open one.

**Comment by jlblancoc** [comment]
A quick heads up: ROS 2 distros already have now a working version of gtsam, Yay!

In case you want to add badges to the README or somewhere else on the web, here they are: 

[![Version](https://img.shields.io/ros/v/foxy/gtsam)](https://index.ros.org/search/?term=gtsam) 
[![Version](https://img.shields.io/ros/v/humble/gtsam)](https://index.ros.org/search/?term=gtsam) 
[![Version](https://img.shields.io/ros/v/rolling/gtsam)](https://index.ros.org/search/?term=gtsam) 

Install instructions: `sudo apt install ros-$ROS_DISTRO-gtsam`

Note: this does NOT includes the python wrapper, which anyway is handled probably easier via pip.

Remember, before tagging each new release, to quickly check the dependencies of the package.xml file and update the "version" entry in that file (or ping me). Ideally, the package.xml should be correct before tagging, otherwise fixing version numbers afterwards is a little "hacky".

**Reply by ProfFan** [reply]
I think the Python wrapper would be nice for inclusion as a package, because the ABI is tied to the specific build process...
